packages/react/src/runtime/style-cache.tsx

```diff
@@ -7,18 +7,31 @@ import { ProviderComponent, UseCacheHook } from './types';
  * Cache to hold already used styles.
  * React Context on the server - singleton object on the client.
  */
-const Cache: any = isNodeEnvironment() ? createContext<Record<string, true> | null>(null) : {};
+let ClientCache = {};
+const Cache: any = createContext<Record<string, true> | null>(null) || ClientCache;
 
 if (!isNodeEnvironment()) {
   /**
    * Iterates through all found style elements generated when server side rendering.
-   *
-   * @param cb
    */
   const ssrStyles = document.querySelectorAll<HTMLStyleElement>('style[data-cmpld]');
   for (let i = 0; i < ssrStyles.length; i++) {
+    // Create the client cache for all SSR'd classes
+    const inserted: Record<string, true> = {};
+    const stylesheet = ssrStyles[i];
+    const rulesText = stylesheet.innerText;
+
+    if (rulesText) {
+      rulesText.split('}').forEach((rule: string) => {
```

Comment:
can we use a for loop here? it's faster

an edge case with this implementation will be nested rules, how can we accommodate this? 🤔 

```
@media screen {
  .rule { display: block }
}
```

Reply:
I didn't use a for loop initially because it uses more bytes. Happy to change it. For future reference I assume performance takes precedence over size?